src/routes/versions.rs

```diff
@@ -83,6 +116,29 @@ pub async fn version_get(
         .map_err(|e| ApiError::DatabaseError(e.into()))?;
 
     if let Some(data) = version_data {
+        if !data.accepted {
+            let user = get_user_from_headers(req.headers(), &**pool).await?;
+            let mod_item = database::models::Mod::get(data.mod_id, &**pool)
+                .await
+                .map_err(|e| ApiError::DatabaseError(e.into()))?
+                .ok_or_else(|| {
+                    ApiError::InvalidInputError(
+                        "Attempted to edit version not attached to mod. How did this happen?"
+                            .to_string(),
+                    )
+                })?;
+            let team_member = database::models::TeamMember::get_from_user_id(
+                mod_item.team_id,
+                user.id.into(),
+                &**pool,
+            )
+            .await?;
+
+            if team_member.is_none() && user.role == Role::Developer {
+                return Ok(HttpResponse::NotFound().body(""));
+            }
```

Comment:
`user.role = Role::Developer` is a very bad idea.  If we ever add more roles, they would be automatically treated as moderators.  Instead there should be a method `Role::is_moderator` that determines whether a role should have moderation privileges.